### ISSUE TITLE
Resolving wildcard http proxy paths with event handler

### ISSUE THREAD

**Issue by robdobtob** [awslabs/aws-lambda-powertools-python#728]
We have are building a lambda 'proxy' to intercept and re-route requests through API gateway to another external service. The lambda's are used to perform data transformations on all requests coming in under a specific path
API Gateway path uses the greedy path capture `{proxy+}` and we do not want to have to implement explicit handlers for all the possible greedy path matches

## Expected Behavior
When triggering a GET request to the proxy with url path `https://<API_GATEWAY>/externalService/services/vXX/info`, would expect the `@app.get('/<servicePath+>')` / `get_externalService` route to be triggered with `servicePath` variable containing `/services/vXX/info`

## Current Behavior
Current behaviour returns 404 NOT FOUND

## Possible Solution
Ability for a dynamic route to perform 'greedy' path capture to match the apiGateway path functionality

## Steps to Reproduce (for bugs)

SAM template config for apiGateway lambda proxy
```
APIGatewayGET:
          Type: Api
          Properties:
            Method: GET
            Path: "/externalService/{servicePath+}"
```

Python lambda powertools route
```
app = ApiGatewayResolver(strip_prefixes=[
                         '/externalService'], proxy_type=ProxyEventType.APIGatewayProxyEvent)

@logger.inject_lambda_context
@app.get('/<servicePath+>')
def get_externalService(servicePath):
     # proxies request
    return {'result': 'hello service'}

@logger.inject_lambda_context
def lambda_handler(event, context):
    return app.resolve(event, context)
```

## Environment

* **Powertools version used**: 1.20.1
* **Packaging format (Layers, PyPi)**: Layers
* **AWS Lambda function runtime:** Python 3.8

**Comment by heitorlessa** [comment]
Hi @robdobtob thanks for raising an issue.

I believe this newly documented catch-all/greedy path can solve your issue: https://awslabs.github.io/aws-lambda-powertools-python/develop/core/event_handler/api_gateway/#catch-all-routes

It's in the stages docs until our next release this week but the functionality is already available.

Let us know if this helps.

Thank you

**Comment by heitorlessa** [comment]
Hello ;) This is now available in the latest release 1.21.0: https://awslabs.github.io/aws-lambda-powertools-python/develop/core/event_handler/api_gateway/#catch-all-routes

Please feel free to reopen if this wasn't it
